src/mongo_logger.ts

```diff
@@ -329,15 +358,36 @@ function attachCommandFields(
 
 function attachConnectionFields(
   log: Record<string, any>,
-  connectionPoolEvent: ConnectionPoolMonitoringEvent
+  connectionOrSDAMEvent: ConnectionPoolMonitoringEvent | ServerOpeningEvent | ServerClosedEvent
```

Comment:
```suggestion
  event: ConnectionPoolMonitoringEvent | ServerOpeningEvent | ServerClosedEvent
```

I imagine there's more events that may end up here? I think we can just simplify

Reply:
Sounds good, I've changed it locally, and will push one commit altogether.